clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/KafkaInputDictTrainer.java

```diff
@@ -151,12 +195,28 @@ public synchronized byte[] trainDict() {
             mapperValue = recordReader.createValue();
           }
           while (recordReader.next(mapperKey, mapperValue)) {
-            byte[] value = ByteUtils.extractByteArray(mapperValue.value);
-            currentFilledSize += value.length;
+            /**
+             * We can only decompress full compressed value here.
+             * If the source version is using {@link CompressionStrategy.NO_OP}, the value will be passed to the dict directly.
+             * If the source version is using other compression strategies, since we can only decompress the full value here, and the
+             * chunked values will be skipped.
```

Comment:
That is fine for now, but in some edge case it might lead to suboptimal compressibility. For example, if chunking kicks in only when a subset of the fields (which contain very large content) are populated, then it would be ideal to train the dictionary on those big fields. It is an edge case so no need to worry about it for now, but maybe having a comment to that effect may be useful.

Reply:
Done